tests/first_stage_test.py

```diff
@@ -1,11 +1,15 @@
+import fcntl
+import os
+import select
 import subprocess
```

Comment:
Try replacing this with`testlib.subprocess`, it provides https://pypi.org/project/subprocess32/ as necessary, which backports the timeout features added in Python 3.3.